src/jdk.httpserver/share/classes/sun/net/httpserver/ServerConfig.java

```diff
@@ -59,60 +59,58 @@ class ServerConfig {
     private static final int maxConnections;
     private static final int maxIdleConnections;
     // The maximum number of request headers allowable
-    private static int maxReqHeaders;
+    private static final int maxReqHeaders;
     // a maximum value for the header list size. This is the
     // names size + values size + 32 bytes per field line
-    private static int maxReqHeadersSize;
+    private static final int maxReqHeadersSize;
     // max time a request or response is allowed to take
     private static final long maxReqTime;
     private static final long maxRspTime;
-    private static long reqRspTimerScheduleMillis;
+    private static final long reqRspTimerScheduleMillis;
     private static final boolean debug;
 
     // the value of the TCP_NODELAY socket-level option
     private static final boolean noDelay;
 
     static {
 
-        idleIntervalMillis = Long.getLong("sun.net.httpserver.idleInterval", DEFAULT_IDLE_INTERVAL_IN_SECS) * 1000;
-        if (idleIntervalMillis <= 0) {
-            idleIntervalMillis = DEFAULT_IDLE_INTERVAL_IN_SECS * 1000;
-        }
+        long providedIdleIntervalSecs = Long.getLong("sun.net.httpserver.idleInterval", DEFAULT_IDLE_INTERVAL_IN_SECS);
+        idleIntervalMillis = Math.multiplyExact(
+                providedIdleIntervalSecs > 0 ? providedIdleIntervalSecs : DEFAULT_IDLE_INTERVAL_IN_SECS,
+                1000);
```

Comment:
That's a change of behaviour - maybe we should keep the original behaviour that was not unreasonable: if the result of the computation overflows, use the implementation default value.

Reply:
Fixed in 518f1eebe2ec0eb2a11f41853c301a61c16e1325.